server/server.go

```diff
@@ -272,18 +267,19 @@ func (s *server) Config() Config {
 
 func (s *server) Shutdown() {
 	s.shutdownOnce.Do(func() {
-		var wg sync.WaitGroup
-
-		wg.Add(1)
+		// Stop the HTTP server gracefully and wait some time for open connections
+		// to be closed. Then force it to be stopped.
 		go func() {
-			// Stop the HTTP server gracefully and wait some time for open connections
-			// to be closed. Then force it to be stopped.
-			s.httpServer.Stop(s.httpServer.Timeout)
-			<-s.httpServer.StopChan()
-			wg.Done()
+			err := s.httpServer.Shutdown(context.Background())
+			if err != nil {
+				s.logger.Log("level", "error", "message", "shutting down server failed", "stack", fmt.Sprintf("%#v", err))
+			}
 		}()
-
-		wg.Wait()
+		<-time.After(3 * time.Second)
```

Comment:
So every single time it waits 3 seconds to drain connections? Not configurable? What if `httpServer.Shutdown` finishes earlier? Why not `time.Sleep`?

Reply:
AFAIK this resembles the implementation as it was already in terms of configurability. No reason why I used the channel instead of the sleep. I am sure this can be more fancy somehow. Anything I can do to make this better now? I am not too fuzzed. For me the biggest win is to get rid of the old lib and to close the decaying issue.

Reply:
We could actually emit a signal via a channel after the shutdown was done and use `time.After` here to either kill on deadline or on signal.

Reply:
Using channel here instead of Sleep is still weird, but fine. I approved that already. Thanks <3